lib/heuristics/scheduling_heuristic.rb

```diff
@@ -400,18 +413,18 @@ def reorder_stops(vrp)
 
           next if result.nil? || !result[:unassigned].empty?
 
-          time_back_to_depot = route_data[:stops].last[:end] + matrix(route_data, route_data[:stops].last[:point_id], route_data[:end_point_id])
-          scheduling_route_time = time_back_to_depot - route_data[:stops].first[:start]
-          solver_route_time = (result[:routes].first[:activities].last[:begin_time] - result[:routes].first[:activities].first[:begin_time]) # last activity is vehicle depot
+          back_to_depot = route_data[:stops].last[:end] +
+                          matrix(route_data, route_data[:stops].last[:point_id], route_data[:vehicle].end_point&.id)
+          scheduling_route_time = back_to_depot - route_data[:stops].first[:start]
+          solver_route_time = (result[:routes].first[:activities].last[:begin_time] -
+                              result[:routes].first[:activities].first[:begin_time]) # last activity is vehicle depot
 
-          next if scheduling_route_time - solver_route_time < @candidate_services_ids.flat_map{ |s| @services_data[s][:durations] }.min ||
-                  result[:routes].first[:activities].collect{ |stop| @indices[stop[:service_id]] }.compact == route_data[:stops].collect{ |s| @indices[s[:id]] } # we did not change our points order
+          next if scheduling_route_time - solver_route_time <
+                  @candidate_services_ids.flat_map{ |s| @services_data[s][:durations] }.min ||
+                  result[:routes].first[:activities].collect{ |stop| @indices[stop[:service_id]] }.compact ==
+                  route_data[:stops].collect{ |s| @indices[s[:id]] } # we did not change our points order
```

Comment:
Can you check other conditions so that they respect the rubocop and they are not splitted in half by the condition?

Reply:
I do not think this is a rule we have defined in rubocoop, I do not have any alert. Maybe we should add it in common rules ?